a4kSubtitles/download.py

```diff
@@ -111,15 +119,31 @@ def __postprocess(core, filepath, lang_code):
             f.write(text.encode(core.utils.default_encoding))
     except: pass
 
+def __copy_sub_local(core, subfile):
+    # Copy the subfile to local.
+    media_name = core.os.path.splitext(core.os.path.basename(core.kodi.xbmc.getInfoLabel('Player.Filename')))[0]
+    sub_name, lang_code, extension = core.os.path.basename(subfile).rsplit(".", 2)
+    file_dest, folder_dest = None, None
+    if core.kodi.get_kodi_setting("subtitles.storagemode") == 0:
+        folder_dest = core.kodi.xbmc.getInfoLabel('Player.Folderpath')
+        file_dest = core.os.path.join(folder_dest, ".".join([media_name, lang_code, extension]))
+    elif core.kodi.get_kodi_setting("subtitles.storagemode") == 1:
+        folder_dest = core.kodi.get_kodi_setting("subtitles.custompath")
+        file_dest = core.os.path.join(folder_dest, ".".join([media_name, lang_code, extension]))
+
+    if file_dest and core.kodi.xbmcvfs.exists(folder_dest):
+        core.kodi.xbmcvfs.copy(subfile, file_dest)
+
 def download(core, params):
     core.logger.debug(lambda: core.json.dumps(params, indent=2))
 
     core.shutil.rmtree(core.utils.temp_dir, ignore_errors=True)
     core.kodi.xbmcvfs.mkdirs(core.utils.temp_dir)
 
     actions_args = params['action_args']
-    lang_code = core.utils.get_lang_id(actions_args['lang'], core.kodi.xbmc.ISO_639_2)
+    lang_code = core.utils.get_lang_id(actions_args['lang'], core.kodi.xbmc.ISO_639_1)
```

Comment:
KODI uses `ISO_639_2` as internal format. It is mentioned several times in the KODI forums and I've tested it before.
Could you provide steps of your own testing that fails with ISO_639_2 and works with ISO_639_1, so that I can verify as well?

Reply:
I thought of going with all most standard one, which it being used by other projects as well.  Kodi does recognized both 2let and 3let

Technically I noticed that my subtitles on media dir are beginning duplicated "with old ones" because old once are 
`2-let` maybe this also due to bazarr and filebot downloading subs 2-let.

I can revert this back however is it possible to make optionable via settings? many would prefer 2let over 3let

Reply:
Well what do we fix exactly here if both are supported? `3-let` should brings more information for the language, and have less collisions. There've been issue reports with Portuguese (Brazil) before and better to be as specific as possible with lang codes.

Reply:
There is nothing were broken here this was changed due to the of 2let are more used out there. So downloaded file doesn't matter for Kodi whether it's 2-let or 3-let Kodi it self would recognize both of them as 3-let. However on other players prioritize "2-let" over "3-let" if both existed to be honest, I don't know what Kodi would prefer of them since both would be converted 3-let

Either way this should works for both so I'll revert this back 😸